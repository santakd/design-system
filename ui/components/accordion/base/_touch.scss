.slds-button.slds-accordion__summary-action {
  line-height: $line-height-button;
}
