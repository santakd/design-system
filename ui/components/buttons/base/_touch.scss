.slds-button {
  @include border-box-line-height($height-tappable);
  font-weight: $font-weight-bold;
}
