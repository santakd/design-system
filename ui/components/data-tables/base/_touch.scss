.slds-th__action {
  padding-right: $square-tappable-xx-small;
}
