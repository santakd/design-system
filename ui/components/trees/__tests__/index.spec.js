/* eslint-env jest */
import React from 'react';

import {
  TreeItem,
  TreeListItem,
  TreeHeader,
  TreeList,
  TreeGroup,
  TreeContainer
} from '../index';

import { MetaTextTree } from '../base/index';

import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);
describe('tree', () => {
  it('renders a base Tree with metatext', () =>
    matchesMarkup(
      <TreeContainer>
        <TreeHeader id="treeheading">Tree Group Header</TreeHeader>
        <TreeList headerId="treeheading">
          <MetaTextTree headerId="treeheading" />
        </TreeList>
      </TreeContainer>
    ));
});

describe('Base Tree - Tree Item', () => {
  it('renders a default tree item', () => matchesMarkup(<TreeItem />));
  it('renders a tree item with a label passed in', () =>
    matchesMarkup(<TreeItem itemLabel="Tree Item Label" />));
  it('renders a tree item with children', () =>
    matchesMarkup(<TreeItem>Tree Item Children</TreeItem>));
  it('renders a tree item with metatext', () =>
    matchesMarkup(<TreeItem metaTextLabel="Tree Item Metatext" />));
  it('renders a tree item with marked term', () =>
    matchesMarkup(
      <TreeItem itemLabel="lightning button" termToMark="butto" />
    ));
  it('renders a tree item with marked term in metatext', () =>
    matchesMarkup(
      <TreeItem
        itemLabel="lightning button"
        metaTextLabel="Some other term"
        termToMark="othe"
      />
    ));
});

describe('Base Tree - Branch Item', () => {
  it('renders a tree branch item with a label passed in', () =>
    matchesMarkup(<TreeItem branchLabel="Tree Branch Label" isBranch />));
  it('renders a tree branch item with metatext', () =>
    matchesMarkup(<TreeItem isBranch metaTextLabel="Tree Item Metatext" />));
});

describe('Base Tree - List Item', () => {
  describe('for Branches', () => {
    it('renders a tree list item for a branch with the required item label', () =>
      matchesMarkup(
        <TreeListItem ariaLevel={1} isBranch itemLabel="Test Branch">
          <TreeItem isBranch itemLabel="Test Branch" />
        </TreeListItem>
      ));
    it('renders a tree list item for a branch with a custom label and metatext', () =>
      matchesMarkup(
        <TreeListItem
          ariaLevel={1}
          isBranch
          itemLabel="Test Branch"
          metaTextLabel="Some Metatext"
        >
          <TreeItem
            isBranch
            itemLabel="Test Branch"
            metaTextLabel="Some Metatext"
          />
        </TreeListItem>
      ));
    it('renders a tree list item for a branch that is selected', () =>
      matchesMarkup(
        <TreeListItem
          ariaLevel={1}
          isBranch
          isSelected
          itemLabel="Test Branch"
          tabIndex={0}
        >
          <TreeItem isBranch itemLabel="Test Branch" />
        </TreeListItem>
      ));
    it('renders a tree list item for a branch that is expanded', () =>
      matchesMarkup(
        <TreeListItem ariaLevel={1} isBranch isExpanded itemLabel="Test Branch">
          <TreeItem isBranch itemLabel="Test Branch" />
        </TreeListItem>
      ));
    it('renders a tree list item at index 0, with no tabIndex, because the branch has a selected item', () =>
      matchesMarkup(
        <TreeListItem
          ariaLevel={1}
          hasSelectedItem
          isBranch
          itemIndex={0}
          itemLabel="Test Branch"
        >
          <TreeItem isBranch itemLabel="Test Branch" />
        </TreeListItem>
      ));
  });
  describe('for individual Items', () => {
    it('renders a default tree list item', () =>
      matchesMarkup(
        <TreeListItem ariaLevel={1}>
          <TreeItem />
        </TreeListItem>
      ));
    it('renders a tree list item with tabIndex of 0', () =>
      matchesMarkup(
        <TreeListItem ariaLevel={1} tabIndex={0}>
          <TreeItem />
        </TreeListItem>
      ));
    it('renders a tree list item with a custom label', () =>
      matchesMarkup(
        <TreeListItem ariaLevel={1} itemLabel="Test Item">
          <TreeItem />
        </TreeListItem>
      ));
    it('renders a tree list item with a custom label and metatext', () =>
      matchesMarkup(
        <TreeListItem
          ariaLevel={1}
          itemLabel="Test Item"
          metaTextLabel="Some Metatext"
        >
          <TreeItem itemLabel="Test Item" metaTextLabel="Some Metatext" />
        </TreeListItem>
      ));
    it('renders a tree list item that is selected', () =>
      matchesMarkup(
        <TreeListItem ariaLevel={1} isSelected tabIndex={0}>
          <TreeItem />
        </TreeListItem>
      ));
  });
});

describe('Base Tree - Tree Header', () => {
  it('renders a default tree header', () =>
    matchesMarkup(<TreeHeader id="treeheading" />));
  it('renders a tree header with children', () =>
    matchesMarkup(<TreeHeader id="treeheading">Tree Group Header</TreeHeader>));
});

describe('Base Tree - Tree List', () => {
  it('renders a default tree list', () =>
    matchesMarkup(<TreeList headerId="treeheading" />));
  it('renders a tree list with children', () =>
    matchesMarkup(
      <TreeList headerId="treeheading">
        <TreeListItem ariaLevel={1}>
          <TreeItem />
        </TreeListItem>
      </TreeList>
    ));
});

describe('Base Tree - Tree Group', () => {
  it('renders a default tree group', () => matchesMarkup(<TreeGroup />));
  it('renders a tree group with children', () =>
    matchesMarkup(
      <TreeGroup>
        <TreeListItem ariaLevel={1}>
          <TreeItem />
        </TreeListItem>
      </TreeGroup>
    ));
});

describe('Base Tree - Tree Container', () => {
  it('renders a default tree container', () =>
    matchesMarkup(<TreeContainer />));
  it('renders a tree container with children', () =>
    matchesMarkup(
      <TreeContainer>
        <TreeHeader id="treeheading">Tree Group Header</TreeHeader>
      </TreeContainer>
    ));
});
