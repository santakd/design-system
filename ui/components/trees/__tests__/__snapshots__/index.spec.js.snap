// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Tree - Branch Item renders a tree branch item with a label passed in 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small"
    tabIndex="-1"
    title="Expand Tree Item"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand Tree Item
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="Tree Item"
    >
      Tree Item
    </span>
  </span>
</div>
`;

exports[`Base Tree - Branch Item renders a tree branch item with metatext 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small"
    tabIndex="-1"
    title="Expand Tree Item"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand Tree Item
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="Tree Item"
    >
      Tree Item
    </span>
    <span
      className="slds-tree__item-meta slds-truncate"
      title="Tree Item Metatext"
    >
      <span
        className="slds-assistive-text"
      >
        :
      </span>
      Tree Item Metatext
    </span>
  </span>
</div>
`;

exports[`Base Tree - List Item for Branches renders a tree list item at index 0, with no tabIndex, because the branch has a selected item 1`] = `
<li
  aria-expanded={false}
  aria-label="Test Branch"
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small"
      tabIndex="-1"
      title="Expand Test Branch"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Test Branch
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Test Branch"
      >
        Test Branch
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for Branches renders a tree list item for a branch that is expanded 1`] = `
<li
  aria-expanded={true}
  aria-label="Test Branch"
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small"
      tabIndex="-1"
      title="Expand Test Branch"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Test Branch
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Test Branch"
      >
        Test Branch
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for Branches renders a tree list item for a branch that is selected 1`] = `
<li
  aria-expanded={false}
  aria-label="Test Branch"
  aria-level={1}
  aria-selected={true}
  role="treeitem"
  tabIndex={0}
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small"
      tabIndex="-1"
      title="Expand Test Branch"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Test Branch
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Test Branch"
      >
        Test Branch
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for Branches renders a tree list item for a branch with a custom label and metatext 1`] = `
<li
  aria-expanded={false}
  aria-label="Test Branch: Some Metatext"
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small"
      tabIndex="-1"
      title="Expand Test Branch"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Test Branch
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Test Branch"
      >
        Test Branch
      </span>
      <span
        className="slds-tree__item-meta slds-truncate"
        title="Some Metatext"
      >
        <span
          className="slds-assistive-text"
        >
          :
        </span>
        Some Metatext
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for Branches renders a tree list item for a branch with the required item label 1`] = `
<li
  aria-expanded={false}
  aria-label="Test Branch"
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small"
      tabIndex="-1"
      title="Expand Test Branch"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Test Branch
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Test Branch"
      >
        Test Branch
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for individual Items renders a default tree list item 1`] = `
<li
  aria-expanded={null}
  aria-label={null}
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
      tabIndex="-1"
      title="Expand Tree Item"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Tree Item
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Tree Item"
      >
        Tree Item
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for individual Items renders a tree list item that is selected 1`] = `
<li
  aria-expanded={null}
  aria-label={null}
  aria-level={1}
  aria-selected={true}
  role="treeitem"
  tabIndex={0}
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
      tabIndex="-1"
      title="Expand Tree Item"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Tree Item
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Tree Item"
      >
        Tree Item
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for individual Items renders a tree list item with a custom label 1`] = `
<li
  aria-expanded={null}
  aria-label={null}
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
      tabIndex="-1"
      title="Expand Tree Item"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Tree Item
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Tree Item"
      >
        Tree Item
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for individual Items renders a tree list item with a custom label and metatext 1`] = `
<li
  aria-expanded={null}
  aria-label={null}
  aria-level={1}
  role="treeitem"
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
      tabIndex="-1"
      title="Expand Test Item"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Test Item
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Test Item"
      >
        Test Item
      </span>
      <span
        className="slds-tree__item-meta slds-truncate"
        title="Some Metatext"
      >
        <span
          className="slds-assistive-text"
        >
          :
        </span>
        Some Metatext
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - List Item for individual Items renders a tree list item with tabIndex of 0 1`] = `
<li
  aria-expanded={null}
  aria-label={null}
  aria-level={1}
  role="treeitem"
  tabIndex={0}
>
  <div
    className="slds-tree__item"
  >
    <button
      aria-hidden="true"
      className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
      tabIndex="-1"
      title="Expand Tree Item"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Expand Tree Item
      </span>
    </button>
    <span
      className="slds-has-flexi-truncate"
    >
      <span
        className="slds-tree__item-label slds-truncate"
        title="Tree Item"
      >
        Tree Item
      </span>
    </span>
  </div>
</li>
`;

exports[`Base Tree - Tree Container renders a default tree container 1`] = `
<div
  className="slds-tree_container"
/>
`;

exports[`Base Tree - Tree Container renders a tree container with children 1`] = `
<div
  className="slds-tree_container"
>
  <h4
    className="slds-tree__group-header"
    id="treeheading"
  >
    Tree Group Header
  </h4>
</div>
`;

exports[`Base Tree - Tree Group renders a default tree group 1`] = `
<ul
  role="group"
/>
`;

exports[`Base Tree - Tree Group renders a tree group with children 1`] = `
<ul
  role="group"
>
  <li
    aria-expanded={null}
    aria-label={null}
    aria-level={1}
    role="treeitem"
  >
    <div
      className="slds-tree__item"
    >
      <button
        aria-hidden="true"
        className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
        tabIndex="-1"
        title="Expand Tree Item"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Expand Tree Item
        </span>
      </button>
      <span
        className="slds-has-flexi-truncate"
      >
        <span
          className="slds-tree__item-label slds-truncate"
          title="Tree Item"
        >
          Tree Item
        </span>
      </span>
    </div>
  </li>
</ul>
`;

exports[`Base Tree - Tree Header renders a default tree header 1`] = `
<h4
  className="slds-tree__group-header"
  id="treeheading"
/>
`;

exports[`Base Tree - Tree Header renders a tree header with children 1`] = `
<h4
  className="slds-tree__group-header"
  id="treeheading"
>
  Tree Group Header
</h4>
`;

exports[`Base Tree - Tree Item renders a default tree item 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
    tabIndex="-1"
    title="Expand Tree Item"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand Tree Item
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="Tree Item"
    >
      Tree Item
    </span>
  </span>
</div>
`;

exports[`Base Tree - Tree Item renders a tree item with a label passed in 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
    tabIndex="-1"
    title="Expand Tree Item Label"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand Tree Item Label
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="Tree Item Label"
    >
      Tree Item Label
    </span>
  </span>
</div>
`;

exports[`Base Tree - Tree Item renders a tree item with children 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
    tabIndex="-1"
    title="Expand Tree Item"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand Tree Item
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="Tree Item"
    >
      Tree Item
    </span>
  </span>
  Tree Item Children
</div>
`;

exports[`Base Tree - Tree Item renders a tree item with marked term 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
    tabIndex="-1"
    title="Expand lightning button"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand lightning button
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="lightning button"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "lightning <mark>butto</mark>n",
          }
        }
      />
    </span>
  </span>
</div>
`;

exports[`Base Tree - Tree Item renders a tree item with marked term in metatext 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
    tabIndex="-1"
    title="Expand lightning button"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand lightning button
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="lightning button"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "lightning button",
          }
        }
      />
    </span>
    <span
      className="slds-tree__item-meta slds-truncate"
      title="Some other term"
    >
      <span
        className="slds-assistive-text"
      >
        :
      </span>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some <mark>othe</mark>r term",
          }
        }
      />
    </span>
  </span>
</div>
`;

exports[`Base Tree - Tree Item renders a tree item with metatext 1`] = `
<div
  className="slds-tree__item"
>
  <button
    aria-hidden="true"
    className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
    tabIndex="-1"
    title="Expand Tree Item"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Expand Tree Item
    </span>
  </button>
  <span
    className="slds-has-flexi-truncate"
  >
    <span
      className="slds-tree__item-label slds-truncate"
      title="Tree Item"
    >
      Tree Item
    </span>
    <span
      className="slds-tree__item-meta slds-truncate"
      title="Tree Item Metatext"
    >
      <span
        className="slds-assistive-text"
      >
        :
      </span>
      Tree Item Metatext
    </span>
  </span>
</div>
`;

exports[`Base Tree - Tree List renders a default tree list 1`] = `
<ul
  aria-labelledby="treeheading"
  className="slds-tree"
  role="tree"
/>
`;

exports[`Base Tree - Tree List renders a tree list with children 1`] = `
<ul
  aria-labelledby="treeheading"
  className="slds-tree"
  role="tree"
>
  <li
    aria-expanded={null}
    aria-label={null}
    aria-level={1}
    role="treeitem"
  >
    <div
      className="slds-tree__item"
    >
      <button
        aria-hidden="true"
        className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
        tabIndex="-1"
        title="Expand Tree Item"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Expand Tree Item
        </span>
      </button>
      <span
        className="slds-has-flexi-truncate"
      >
        <span
          className="slds-tree__item-label slds-truncate"
          title="Tree Item"
        >
          Tree Item
        </span>
      </span>
    </div>
  </li>
</ul>
`;

exports[`tree renders a base Tree with metatext 1`] = `
<div
  className="slds-tree_container"
>
  <h4
    className="slds-tree__group-header"
    id="treeheading"
  >
    Tree Group Header
  </h4>
  <ul
    aria-labelledby="treeheading"
    className="slds-tree"
    role="tree"
  >
    <li
      aria-expanded={null}
      aria-label={null}
      aria-level={1}
      role="treeitem"
      tabIndex={0}
    >
      <div
        className="slds-tree__item"
      >
        <button
          aria-hidden="true"
          className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
          tabIndex="-1"
          title="Expand Tree Item"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Expand Tree Item
          </span>
        </button>
        <span
          className="slds-has-flexi-truncate"
        >
          <span
            className="slds-tree__item-label slds-truncate"
            title="Tree Item"
          >
            Tree Item
          </span>
        </span>
      </div>
    </li>
    <li
      aria-expanded={true}
      aria-label="Tree Branch"
      aria-level={1}
      role="treeitem"
    >
      <div
        className="slds-tree__item"
      >
        <button
          aria-hidden="true"
          className="slds-button slds-button_icon slds-m-right_x-small"
          tabIndex="-1"
          title="Expand Tree Branch"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Expand Tree Branch
          </span>
        </button>
        <span
          className="slds-has-flexi-truncate"
        >
          <span
            className="slds-tree__item-label slds-truncate"
            title="Tree Branch"
          >
            Tree Branch
          </span>
        </span>
      </div>
      <ul
        role="group"
      >
        <li
          aria-expanded={null}
          aria-label={null}
          aria-level={2}
          role="treeitem"
        >
          <div
            className="slds-tree__item"
          >
            <button
              aria-hidden="true"
              className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
              tabIndex="-1"
              title="Expand Tree Item"
            >
              <svg
                aria-hidden={true}
                className="slds-button__icon slds-button__icon_small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
                />
              </svg>
              <span
                className="slds-assistive-text"
              >
                Expand Tree Item
              </span>
            </button>
            <span
              className="slds-has-flexi-truncate"
            >
              <span
                className="slds-tree__item-label slds-truncate"
                title="Tree Item"
              >
                Tree Item
              </span>
              <span
                className="slds-tree__item-meta slds-truncate"
                title="Tree Item Metatext with a Really Really Long Name That Should Truncate"
              >
                <span
                  className="slds-assistive-text"
                >
                  :
                </span>
                Tree Item Metatext with a Really Really Long Name That Should Truncate
              </span>
            </span>
          </div>
        </li>
      </ul>
    </li>
    <li
      aria-expanded={false}
      aria-label="Tree Branch: Tree Branch Metatext"
      aria-level={1}
      role="treeitem"
    >
      <div
        className="slds-tree__item"
      >
        <button
          aria-hidden="true"
          className="slds-button slds-button_icon slds-m-right_x-small"
          tabIndex="-1"
          title="Expand Tree Branch"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Expand Tree Branch
          </span>
        </button>
        <span
          className="slds-has-flexi-truncate"
        >
          <span
            className="slds-tree__item-label slds-truncate"
            title="Tree Branch"
          >
            Tree Branch
          </span>
          <span
            className="slds-tree__item-meta slds-truncate"
            title="Tree Branch Metatext"
          >
            <span
              className="slds-assistive-text"
            >
              :
            </span>
            Tree Branch Metatext
          </span>
        </span>
      </div>
      <ul
        role="group"
      >
        <li
          aria-expanded={null}
          aria-label={null}
          aria-level={2}
          role="treeitem"
        >
          <div
            className="slds-tree__item"
          >
            <button
              aria-hidden="true"
              className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
              tabIndex="-1"
              title="Expand Tree Item"
            >
              <svg
                aria-hidden={true}
                className="slds-button__icon slds-button__icon_small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
                />
              </svg>
              <span
                className="slds-assistive-text"
              >
                Expand Tree Item
              </span>
            </button>
            <span
              className="slds-has-flexi-truncate"
            >
              <span
                className="slds-tree__item-label slds-truncate"
                title="Tree Item"
              >
                Tree Item
              </span>
            </span>
          </div>
        </li>
      </ul>
    </li>
    <li
      aria-expanded={null}
      aria-label={null}
      aria-level={1}
      role="treeitem"
    >
      <div
        className="slds-tree__item"
      >
        <button
          aria-hidden="true"
          className="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
          tabIndex="-1"
          title="Expand Tree Item with a Really Really Long Name That Should Truncate"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Expand Tree Item with a Really Really Long Name That Should Truncate
          </span>
        </button>
        <span
          className="slds-has-flexi-truncate"
        >
          <span
            className="slds-tree__item-label slds-truncate"
            title="Tree Item with a Really Really Long Name That Should Truncate"
          >
            Tree Item with a Really Really Long Name That Should Truncate
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
